src/libgit2/index.c

```diff
@@ -2849,8 +2850,9 @@ static int parse_index(git_index *index, const char *buffer, size_t buffer_size)
 	/*
 	 * SHA-1 or SHA-256 (depending on the repository's object format)
 	 * over the content of the index file before this checksum.
+	 * Note: checksum may be 0 if index.skipHash is set to true.
```

Comment:
My reading of this suggested that `index.skipHash` was supported during read; I'd suggest we change the messaging a bit to when we _encounter_ a null hash.

```suggestion
	 * Note: checksum may be 0 if the index was written by a client
	 * where index.skipHash was set to true.
```